pkg/service/service.go

```diff
@@ -134,15 +139,30 @@ func NewService(monitorNotify monitorNotify) *Service {
 	if option.Config.EnableHealthCheckNodePort {
 		localHealthServer = healthserver.New()
 	}
-	return &Service{
+	svc := &Service{
 		svcByHash:       map[string]*svcInfo{},
 		svcByID:         map[lb.ID]*svcInfo{},
 		backendRefCount: counter.StringCounter{},
 		backendByHash:   map[string]*lb.Backend{},
 		monitorNotify:   monitorNotify,
 		healthServer:    localHealthServer,
 		lbmap:           &lbmap.LBBPFMap{},
+		clock:           clock.RealClock{},
 	}
+	svc.lastUpdatedTs.Store(svc.clock.Now())
+	return svc
+}
+
+func (s *Service) GetLastUpdatedTs() time.Time {
+	var ts time.Time
+	if val := s.lastUpdatedTs.Load(); val != nil {
```

Comment:
Curious: I believe this check is superfluous because we initialize lastUpdateTs when we create it?

Reply:
Addressed review comments.

Reply:
Addressed review comments.

Reply:
Considering that someone might accidentally move the initialization of `lastUpdatedTs` which currently is happening in `NewService()` function, I'd add the `ts, ok := val.(time.Time); if !ok { ... }` check.